samples/nrf9160/http_application_update/src/main.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <stdint.h>
+#include <zephyr.h>
+#include <flash.h>
+#include <bsd.h>
+#include <nrf_socket.h>
+#include <download_client.h>
+#include <logging/log.h>
+#include <gpio.h>
+#include <flash_map.h>
+#include <dfu/mcuboot.h>
+
+LOG_MODULE_REGISTER(main);
+
+#define LED_PORT	LED0_GPIO_CONTROLLER
+
+static struct	device	*flash_dev;
+static K_THREAD_STACK_DEFINE(dfu_thread_stack, 1024);
+static bool is_flash_page_erased[FLASH_PAGE_MAX_CNT];
+
+static struct	k_thread dfu_thread;
+static bool		start_dfu;
+static struct	gpio_callback gpio_cb;
+
+static int app_download_client_event_handler(struct download_client *const dfu,
+			enum download_client_evt event, u32_t status);
+
+
+static struct download_client dfu = {
```

Comment:
Can be `const`.

Reply:
Fixed now.

Reply:
Removed const again to be compatible with download client library.

Reply:
My mistake. The struct cannot be `const`.